src/pages/groupComparison/UpSet.tsx

```diff
@@ -0,0 +1,489 @@
+import * as React from 'react';
+import { observer, Observer } from "mobx-react";
+import { VictoryChart, VictoryAxis, VictoryBar, VictoryScatter, VictoryLine, VictoryLabel } from 'victory';
+import { computed, observable } from 'mobx';
+import * as _ from 'lodash';
+import CBIOPORTAL_VICTORY_THEME, { axisTickLabelStyles } from 'shared/theme/cBioPoralTheme';
+import autobind from 'autobind-decorator';
+import { getCombinations, ComparisonGroup } from './GroupComparisonUtils';
+import { getTextWidth } from 'shared/lib/wrapText';
+import { tickFormatNumeral } from 'shared/components/plots/TickUtils';
+import Timer = NodeJS.Timer;
+import ScatterPlotTooltip from 'shared/components/plots/ScatterPlotTooltip';
+import { joinNames, blendColors } from './OverlapUtils';
```

Comment:
blendColors was only designed to handle up to 3 groups, you need to rewrite it to iteratively find halfway point between arbitrary number. Let's talk if that doesn't make sense

Reply:
done!